### PR TITLE
Switch important CI from CUDA 9 to 10.1

### LINKED ISSUE #31427: Should we expose CircleCI CUDA10 tests as XImportant and run on PRs?
This comes when working on https://github.com/pytorch/pytorch/pull/30898

**Long story short:**
`torch.addmv` for `torch.half` has a different code path from other dtypes. Due to an issue on CUDA9 on old hardware (https://github.com/pytorch/pytorch/issues/31006) I have to disable the zero-sized tensor test for `torch.addmv` + `torch.half` on CUDA9. This means this specific code path is not covered by any test in our CI for PR. The skipped test is:
```python
@skipCUDAIfRocm
@dtypesIfCUDA(*(torch.float, torch.double) +
              # This test is disabled on CUDA 9, due to:
              # See: https://github.com/pytorch/pytorch/issues/31006
              ((torch.half,) if torch.version.cuda and float(torch.version.cuda) >= 10.0 else ()))
@dtypes(*(set(torch.testing.get_all_dtypes()) - {torch.half, torch.bool}))
def test_blas_alpha_beta_empty(self, device, dtype):
```

This case seems to be a very edge case. I don't know if it worth to enable CUDA 10 test on PRs, or just switch from CUDA 9 to CUDA 10 test on PRs.

cc: @ngimel @ezyang 
cc: @VitalyFedyunin 

cc @ezyang @ngimel

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Fixes https://github.com/pytorch/pytorch/issues/31427

### CODE REVIEW

**Comment by kostmo** [comment]
## :pill: CircleCI build failures summary and remediations
As of commit cc0aa92c:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=cc0aa92c559c6e3caa854b6de331eddf9b4342cb).

### 1 failure *not* recognized by patterns:
| Job | Step | Status |
| --- | --- | --- |
| CircleCI <sup>Build Error</sup> | <sup>Config Processing Error (Don't rerun)</sup> | New in PR |


---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

File: .circleci/cimodel/data/pytorch_build_data.py [hunk 1]
```diff
@@ -39,14 +39,16 @@
                 # and
                 # https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/build.sh#L153
                 # (from https://github.com/pytorch/pytorch/pull/17323#discussion_r259453144)
+                X("3.6"),
```
Comment (zasdfgbnm) [inline .circleci/cimodel/data/pytorch_build_data.py]: We will lose libtorch test fon CUDA 9 both on PR and on master branch. Is that OK?

Comment (ezyang) [reply .circleci/cimodel/data/pytorch_build_data.py]: Yeah, that seems fine, as long as libtorch is being tested on CUDA 10.1

**Review by ezyang — APPROVED** [review]
Can you manually inspect the CI results and make sure they look kosher? Specifically, make sure we are not skipping things we shouldn't. Once you do so I'll start landing.

**Comment by zasdfgbnm** [comment]
I will wait for https://github.com/pytorch/pytorch/pull/31964 then compare it with this.

**Comment by zasdfgbnm** [comment]
multigpu test and slow test seems to be missing... I will look into it.

**Comment by ezyang** [comment]
Let me know when you're good to go

**Comment by ezyang** [comment]
With your most recent changes you're going to need to rebuild the docker images

**Comment by zasdfgbnm** [comment]
@ezyang Do you mean to rebuild and push it somewhere? BTW I found some cuda8 things in `.circleci/docker/build.sh` could these be removed?

**Comment by ezyang** [comment]
Yes, follow the instructions at https://github.com/pytorch/pytorch/wiki/Docker-image-build-on-CircleCI

> BTW I found some cuda8 things in .circleci/docker/build.sh could these be removed?

Yep, go for it.

**Comment by zasdfgbnm** [comment]
@ezyang The CI logic is much more complicated than what I originally thought to be. I will write a list of what I checked and what I didn't check (clicking into all the 62 tests and read the log is time-consuming and not necessary), I will let you know when I am ready and please review the list before landing.

**Comment by zasdfgbnm** [comment]
[WIP] List of things that I checked and I thing there is no need to check. Checking is against https://github.com/pytorch/pytorch/pull/31964:
- ✔️ There are 62 CI checks, for all these checks, I checked that their names match each other except a replacement of `9 -> 10.1`
- ❌ I didn't click into checks that are
  - `binary_*`
  - `caffe2_*`
  - `*android*`
  - `*py2*`
  - `*macos*`
  - `*parallelnative*`
  - `*paralleltbb*`
  - `*windows*`
  - `*xla*`
  - `*rocm*` 
  - `pytorch_linux_xenial_py*` (cpu only checks)

**Comment by ezyang** [comment]
Your checks are more than sufficient. Thank you for looking.

**Comment by ezyang** [comment]
Looks like katex is still being annoying:

```
Jan 11 01:06:58 Exception occurred:
Jan 11 01:06:58   File "/opt/conda/lib/python3.6/subprocess.py", line 1364, in _execute_child
Jan 11 01:06:58     raise child_exception_type(errno_num, err_msg, err_filename)
Jan 11 01:06:58 FileNotFoundError: [Errno 2] No such file or directory: 'katex': 'katex'
Jan 11 01:06:58 The full traceback has been saved in /tmp/sphinx-err-unpcocz5.log, if you want to report the issue to the developers.
Jan 11 01:06:58 Please also report this if it was a user error, so that a better error message can be provided next time.
Jan 11 01:06:58 A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Jan 11 01:06:59 Makefile:38: recipe for target 'html' failed
Jan 11 01:06:59 make: *** [html] Error 2
```

**Comment by zasdfgbnm** [comment]
@ezyang Yes, I was waiting for https://github.com/pytorch/pytorch/pull/32013 then rebuild docker

**Comment by zasdfgbnm** [comment]
@ezyang This is ready to go. Just one question: do I need to update https://github.com/pytorch/ossci-job-dsl/blob/master/src/main/groovy/ossci/pytorch/DockerVersion.groovy ? If yes, I can open a PR on Tuesday.

**Comment by ezyang** [comment]
> This is ready to go. Just one question: do I need to update https://github.com/pytorch/ossci-job-dsl/blob/master/src/main/groovy/ossci/pytorch/DockerVersion.groovy ? If yes, I can open a PR on Tuesday.

Hrm. I *think* this is dead now, but I need to check. Let's get this in for now.

**Comment by ezyang** [comment]
Verified this is dead.

**Pull request closed without merging**